cost-model/src/transaction_cost.rs

```diff
@@ -1,10 +1,109 @@
-use solana_sdk::pubkey::Pubkey;
+use {crate::block_cost_limits, solana_sdk::pubkey::Pubkey};
+
+/// TransactionCost is used to represent resources required to process
+/// a transaction, denominated in CU (eg. Compute Units).
+/// Resources required to process a regular transaction often include
+/// an array of variables, such as execution cost, loaded bytes, write
+/// lock and read lock etc.
+/// Vote has a simpler and pre-determined format, it's cost structure
+/// can be simpler, calculation quicker.
+#[derive(Debug)]
+pub enum TransactionCost {
+    Vote { writable_accounts: Vec<Pubkey> },
+    Transaction(UsageCostDetails),
+}
+
+impl TransactionCost {
+    pub fn sum(&self) -> u64 {
+        match self {
+            Self::Vote { writable_accounts } => {
+                let num_of_signatures = writable_accounts.len() as u64;
+
+                solana_vote_program::vote_processor::DEFAULT_COMPUTE_UNITS
+                    .saturating_add(
+                        block_cost_limits::SIGNATURE_COST.saturating_mul(num_of_signatures),
+                    )
+                    .saturating_add(
+                        block_cost_limits::WRITE_LOCK_UNITS.saturating_mul(num_of_signatures),
+                    )
+            }
+            Self::Transaction(usage_cost) => usage_cost.sum(),
+        }
+    }
+
+    pub fn bpf_execution_cost(&self) -> u64 {
+        match self {
+            Self::Vote { .. } => 0,
+            Self::Transaction(usage_cost) => usage_cost.bpf_execution_cost,
+        }
+    }
+
+    pub fn is_simple_vote(&self) -> bool {
```

Comment:
🤔 I'm not sure about a base assumption here. We call them vote and "non-vote" transactions but I'm not 100% sure if that's accurate. From what I recall, we actually have "**simple** vote" and non-simple-vote. i.e. I'm not aware there's anything preventing a vote ix with other ixs going through the "non-vote" path.

If we separate the costs, we should be sure that vote ixs aren't allowed through the other paths.

Reply:
Maybe @AshwinSekar would know more about this.

Reply:
Good question for @AshwinSekar 😄 I didn't think about it, but yea i dont think we fail a tx if it includes a vote instruction. Interesting to know if that vote embedded in a normal transaction will be accounted for. 

But in any case, here we give "simple vote" is fast track. if there is vote ix in a normal transaction, then that transaction's cost will be counted accordingly, including the cost of vote ix. It just go down normal path.

Reply:
Yes technically all votes sent by the stock validator will include the flag on packet indicating that it is a simple vote, however in theory someone could craft a tx without that flag set.

Reply:
Is this something we want to continue supporting?
My concern is that some time in the future some economic change happens without consideration of votes, and it becomes cheaper to send votes via non-vote channels.
Then our vote threads essentially become useless

Reply:
Personally i think it shouldn't be supported. I think when i checked a couple years ago everyone was sending votes as single tx with the flag set. I'm fine for rejecting votes that don't fit this criteria.

It does reject some freedom, but i can't imagine a use case where someone would need to bundle a vote ix with something else.

Reply:
my preference would be: 
1. vote tx must be "simple vote tx" as defined today (1 or 2 sig, legacy msg, etc)
2. any other transaction contains vote instruction should fail.

I'm kind of surprised that why #2 is not already enforced.

Reply:
Protocol would need to be updated so we reject these txs on replay.
Definitely needs a feature-gate. I guess since it's a protocol change, should this go through a SIMD?

Reply:
yea, that would be a separate discussion for sure.

This PR is to simplify "simple-vote" CU cost calc, since it is pretty restrictive and static for being a 'simple-vote'.